fuseops/ops.go

```diff
@@ -654,6 +654,37 @@ type ReadFileOp struct {
 	OpContext OpContext
 }
 
+// Vectored read - same as ReadFileOp, but the buffer isn't provided by the library.
+// The file system returns a list of slices instead.
+type VectoredReadOp struct {
+	// The file inode that we are reading, and the handle previously returned by
+	// CreateFile or OpenFile when opening that inode.
+	Inode  InodeID
+	Handle HandleID
+
+	// The offset within the file at which to read.
+	Offset int64
+
+	// The size of the read.
+	Size int64
+
+	// Set by the file system: data to send back to the client.
+	Data [][]byte
```

Comment:
I wonder if we could add a new field (e.g. VectoredData, and Size) to the existing ReadFileOp type?

If UseVectoredRead is enabled, VectoredData is used, otherwise Dst is used. Size can be set in either case.

Reply:
Maybe, in fact that came first to my mind, but then I decided that a separate operation was more explicit, i.e. you get a ReadFileOp and you're 100% sure that it doesn't have empty .Dst. Also if both variants are merged in the same op then Size should probably be set for both variants and someone may begin using that field in the future... But that's up to you, I think I can merge them back if you insist :)

Reply:
Can you (perhaps in a separate PR) merge them back so that we can see what that looks like? In particular, changes to the intreface (fuseutil/file_system.go) also break compatibility for users.

If we end up going with two separate types, we need to factor out the common fields into a separate struct.